src/Testcontainers/Configurations/TestcontainersSettings.cs

```diff
@@ -25,7 +25,14 @@ public static class TestcontainersSettings
     private static readonly ICustomConfiguration EnvironmentConfiguration = new EnvironmentConfiguration();
 
     private static readonly IDockerEndpointAuthenticationConfiguration DockerEndpointAuthConfig =
-      new IDockerEndpointAuthenticationProvider[] { new EnvironmentEndpointAuthenticationProvider(), new NpipeEndpointAuthenticationProvider(), new UnixEndpointAuthenticationProvider() }
+      new IDockerEndpointAuthenticationProvider[]
```

Comment:
This solution brings an ideato me. Do we want to open it for 3rd party customization? Like to add other providers from users. 
Maybe same approach can be used for custom configuration.

Reply:
Yep, we can do that in the future. Until now, other devs can set the provider via `TestcontainersSettings.OS.DockerEndpointAuthConfig` too.